### PR TITLE
Fixes hivemind exploits of building without weeds

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
hiveminds will now back off when they try to build weeds with no weeds nearby, they'll go either to another weed in 7 tiles range or go to their core, also did same with moving with no weeds, if there are any weeds in 7 tiles range, they'll go back to that weed instead of straight to core

## Why It's Good For The Game
Removes stinky annoying bug.

## Changelog
:cl:
tweak: when hiveminds move with no weed nearby they wont go straight to core, they'll go to the closest weed in 7 tiles.
fix: hiveminds are not able to build weeds with no weeds nearby anymore.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 1]
```diff
@@ -92,6 +92,18 @@
 /datum/action/xeno_action/plant_weeds/slow
 	cooldown_timer = 12 SECONDS
 
+/datum/action/xeno_action/plant_weeds/slow/action_activate()
+	var/mob/living/carbon/xenomorph/hivemind/hiveminde = owner
+	if(hiveminde)
+		if(!locate(/obj/effect/alien/weeds) in range(1, owner.loc))
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:98]: why do you chack for owner

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 2]
```diff
@@ -92,6 +92,18 @@
...
+			var/obj/effect/alien/weeds/nearbies = locate() in range(7, hiveminde.loc)
+			if(nearbies)
+				hiveminde.forceMove(get_turf(nearbies))
+			else
+				hiveminde.forceMove(get_turf(hiveminde.core))
+			to_chat(hiveminde, "<span class='xenonotice'>you can't place weeds with no weeds nearby, you got moved to [nearbies ? "the closest weed" : "your core"].</span>")
+			return
+ 	..()
 // Choose Resin
 /datum/action/xeno_action/choose_resin
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:108]:
```suggestion
 	return ..()
 
// Choose Resin
/datum/action/xeno_action/choose_resin
```

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm [hunk 3]
```diff
@@ -46,12 +46,16 @@
...
+		var/obj/effect/alien/weeds/nearby = locate() in range(1, loc)
 		if(!nearby)
-			// If we run out of weeds just teleport to some random weeds.
-			forceMove(get_turf(core))
+			var/obj/effect/alien/weeds/nearbiest = locate() in  range(7, loc)
+			if(nearbiest) 	// If we run out of weeds just teleport to some random weeds that aint so far away
+				forceMove(get_turf(nearbiest))
+			else
+				forceMove(get_turf(core))
+			to_chat(src, "<span class='xenonotice'>you had no weeds nearby, you got moved to [nearbiest ? "the closest weed" : "your core"].")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:58]:
```suggestion
			to_chat(src, "<span class='xenonotice'>You had no weeds nearby, you got moved to [nearbiest ? "the closest weed" : "your core"].")
```

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm [hunk 4]
```diff
@@ -46,12 +46,16 @@
...
-		var/obj/effect/alien/weeds/nearby = locate() in range("3x3", loc)
+		var/obj/effect/alien/weeds/nearby = locate() in range(1, loc)
 		if(!nearby)
-			// If we run out of weeds just teleport to some random weeds.
-			forceMove(get_turf(core))
+			var/obj/effect/alien/weeds/nearbiest = locate() in  range(7, loc)
+			if(nearbiest) 	// If we run out of weeds just teleport to some random weeds that aint so far away
+				forceMove(get_turf(nearbiest))
+			else
+				forceMove(get_turf(core))
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:57]: Not a fan of this locate() spam Can you just revert it to previous functionality here

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities.dm (hunk 1 above)
Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:98]: me dumb, me fix

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm (hunk 4 above)
Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:57]: sure

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 5]
```diff
@@ -92,6 +92,19 @@
...
+	if(hiveminde)
+		if(!locate(/obj/effect/alien/weeds) in range(1, hiveminde.loc))
+			var/obj/effect/alien/weeds/nearbies = locate() in range(7, hiveminde.loc)
+			if(nearbies)
+				hiveminde.forceMove(get_turf(nearbies))
+			else
+				hiveminde.forceMove(get_turf(hiveminde.core))
+			to_chat(hiveminde, "<span class='xenonotice'>you can't place weeds with no weeds nearby, you got moved to [nearbies ? "the closest weed" : "your core"].</span>")
+			return
+ 	return ..()
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:106]:
```suggestion
	var/mob/living/carbon/xenomorph/hivemind/hiveminde = owner
	if(!locate(/obj/effect/alien/weeds) in range(1, hiveminde.loc))
		var/obj/effect/alien/weeds/nearbies = locate() in range(7, hiveminde.loc)
		if(nearbies)
			hiveminde.forceMove(get_turf(nearbies))
		else
			hiveminde.forceMove(get_turf(hiveminde.core))
		to_chat(hiveminde, "<span class='xenonotice'>you can't place weeds with no weeds nearby, you got moved to [nearbies ? "the closest weed" : "your core"].</span>")
			return
 	return ..()
```
?

Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:106]: huh ?

Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: if(hiveminde) to make sure its a hiveminde if someone ever wants to add that slow ability to anything else(shoulnd't) and avoid runtimes by not having hiveminde

Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: A) Hiveminde is not istyped
B) something like this shouldnt be istyping in the first place

Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: it doens't need istyping because it already is the type, but i can do that ye

Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: Owner is always set why are you doing this

Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: for hivemind.core

Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: im aware but if(hiveminde) will always be true so theres no point to having it

Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: ok fair

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**